src/components/vm/consoles/vnc.tsx

```diff
@@ -481,6 +512,8 @@ export class VncActive extends React.Component<VncActiveProps, VncActiveState> {
 
     onSecurityFailure(reason: string | undefined) {
         console.info('Security failure:', reason || "unknown reason");
+        if (reason)
+            this.props.state.setDisconnected(reason);
```

Comment:
This condition looks a bit dubious: If we get here, isn't that always a connection failure, even if we don't have a reason? I.e. should this unconditionally call setDisconnected()?

Reply:
Good observation!

It happens to be so that we _also_ get a `onDisconnected` event, and I made it so that `onSecurityFailure` and `onDisconnected` can be called in any order. But a comment is certainly in order.